### ISSUE TITLE
💡 [REQUEST] - Tutorial on deep survival analysis using PyTorch & TorchSurv

### ISSUE THREAD

**Issue by tcoroller** [pytorch/tutorials#2978]
### 🚀 Describe the improvement or the new tutorial

[`TorchSurv`](https://github.com/Novartis/torchsurv) is a Python package that serves as a companion tool to perform deep survival modeling within the `PyTorch` environment. Unlike existing libraries that impose specific parametric forms on users, `TorchSurv` enables the use of custom `PyTorch`-based deep survival models. With its lightweight design, minimal input requirements, full `PyTorch` backend, and freedom from restrictive survival model parameterizations, `TorchSurv` facilitates efficient survival model implementation, particularly beneficial for high-dimensional input data scenarios.

In this tutorial, we want to introduce how to easily use our package, from `loss functions` (Weibull and Cox model), `evaluation metrics` (concordance-index, AUC, Brier score) and `statistical tools` (Kaplan-Meier, estimator). This will enable `Pytorch` users to **develop true survival model by changing few lines of code** while using their favorite deep learning framework!

### Existing tutorials on this topic

The tutorial will be adapted from our existing documentations:
* [introduction to TorchSurv](https://opensource.nibr.com/torchsurv/notebooks/introduction.html)
* [survival example with MNIST](https://opensource.nibr.com/torchsurv/notebooks/momentum.html)

### Additional context

**category**: `survival analysis`

This work was made as part of the collaboration research between the `FDA` and `Novartis`

Further read:
* Our preprint manuscript can be found [here](https://arxiv.org/abs/2404.10761). 
* Features comparison between best `R` and `Python` packages can be found in [this section](https://opensource.nibr.com/torchsurv/index.html#related-packages)
* Performance benchmarks and evaluations can be found [here](https://opensource.nibr.com/torchsurv/benchmarks.html)

**Comment by tcoroller** [comment]
Hello @svekars, I hope you are doing well! Do you have any follow-up on this?

thanks!

**Comment by tcoroller** [comment]
Hello @svekars and @albanD, would you have any thoughts on the request? Our package has been downloaded over 1k last month alone (after only few months since its initial release). We truly believe that this tool will be used by academics as well as industries as there is a gap in reliable package for deep survival learning. We are committed to actively maintain and improve it, and being part of the PyTorch tutorial collection would allow us to reach new users and strengthen our package.

Thanks, 
Thibaud

**Comment by svekars** [comment]
Hi @tcoroller, there is a process to add ecosystem partners. Can you please fill out this form: https://pytorch.org/ecosystem/join and they will follow up with you. After the tool is added to the ecosystem page, we'll be happy to work with you on publishing the tutorials.
